conans/test/unittests/client/generators/virtualrunenv_test.py

```diff
@@ -36,21 +40,19 @@ def test_environment(self):
         self.assertEqual(self.generator.env["DYLD_LIBRARY_PATH"], ["lib1", "lib2"])
 
     def test_scripts(self):
+        content = self.result[self.environment_sh_env]
         self.assertIn('DYLD_LIBRARY_PATH="lib1":"lib2"${DYLD_LIBRARY_PATH+:$DYLD_LIBRARY_PATH}',
-                      self.result[self.activate_sh])
-        self.assertIn('LD_LIBRARY_PATH="lib1":"lib2"${LD_LIBRARY_PATH+:$LD_LIBRARY_PATH}',
-                      self.result[self.activate_sh])
-        self.assertIn('PATH="bin1":"bin2"${PATH+:$PATH}', self.result[self.activate_sh])
+                      content)
+        self.assertIn('LD_LIBRARY_PATH="lib1":"lib2"${LD_LIBRARY_PATH+:$LD_LIBRARY_PATH}', content)
+        self.assertIn('PATH="bin1":"bin2"${PATH+:$PATH}', content)
 
         if platform.system() == "Windows":
-            self.assertIn('DYLD_LIBRARY_PATH=lib1;lib2;%DYLD_LIBRARY_PATH%',
-                          self.result[self.activate_bat])
-            self.assertIn('LD_LIBRARY_PATH=lib1;lib2;%LD_LIBRARY_PATH%',
-                          self.result[self.activate_bat])
-            self.assertIn('PATH=bin1;bin2;%PATH%', self.result[self.activate_bat])
-
-            self.assertIn('$env:DYLD_LIBRARY_PATH = "lib1;lib2;$env:DYLD_LIBRARY_PATH"',
-                          self.result[self.activate_ps1])
-            self.assertIn('$env:LD_LIBRARY_PATH = "lib1;lib2;$env:LD_LIBRARY_PATH"',
-                          self.result[self.activate_ps1])
-            self.assertIn('$env:PATH = "bin1;bin2;$env:PATH"', self.result[self.activate_ps1])
+            content = self.result[self.environment_bat_env]
+            self.assertIn('DYLD_LIBRARY_PATH=lib1;lib2;%DYLD_LIBRARY_PATH%', content)
+            self.assertIn('LD_LIBRARY_PATH=lib1;lib2;%LD_LIBRARY_PATH%', content)
+            self.assertIn('PATH=bin1;bin2;%PATH%', content)
+
+            content = self.result[self.environment_ps1_env]
+            self.assertIn('DYLD_LIBRARY_PATH=lib1;lib2;$env:DYLD_LIBRARY_PATH', content)
```

Comment:
Here the external quotes are being removed. I guess you are certain this will not break paths with spaces?

Reply:
These lines belong to the `.env` file and they are read and used like:

```
    FOR /F "usebackq tokens=1,* delims={{delim}}" %%i IN ("{{ environment_file }}") DO (
        CALL SET "%%i=%%j"
    )
```

which should translate to: `SET "DYLD_LIBRARY_PATH=lib1;lib2;$env:DYLD_LIBRARY_PATH"` (all with quotes again).

For PS1 it is something similar, the line should be quoted back:

```ps1
    foreach ($line in Get-Content "{{ environment_file }}") {
        $var,$value = $line -split '=',2
        $value_expanded = $ExecutionContext.InvokeCommand.ExpandString($value)
        Set-Item env:\\$var -Value "$value_expanded"
    }
```

I think this is already tested and we are not breaking anything, but more testing would be really apreciated.